src/Components/Components/src/Auth/AuthorizeRouteView.cs

```diff
@@ -31,53 +32,61 @@ public AuthorizeRouteView()
             _renderAuthorizedDelegate = authenticateState => renderBaseRouteViewDelegate;
             _renderNotAuthorizedDelegate = authenticationState => RenderContentInDefaultLayout(NotAuthorized(authenticationState));
             _renderAuthorizingDelegate = builder => RenderContentInDefaultLayout(Authorizing);
+            _renderAuthorizeRouteViewCoreDelegate = RenderAuthorizeRouteViewCore;
         }
 
         /// <summary>
         /// The content that will be displayed if the user is not authorized.
         /// </summary>
-        [Parameter] public RenderFragment<AuthenticationState> NotAuthorized { get; set; }
+        [Parameter]
+        public RenderFragment<AuthenticationState> NotAuthorized { get; set; }
 
         /// <summary>
         /// The content that will be displayed while asynchronous authorization is in progress.
         /// </summary>
-        [Parameter] public RenderFragment Authorizing { get; set; }
+        [Parameter]
+        public RenderFragment Authorizing { get; set; }
+
+        [CascadingParameter]
+        private Task<AuthenticationState> ExistingCascadedAuthenticationState { get; set; }
 
         /// <inheritdoc />
         protected override void Render(RenderTreeBuilder builder)
         {
-            // TODO: Consider merging the behavior of CascadingAuthenticationState into this
-            // component (i.e., rendering the CascadingValue directly) to avoid the extra
-            // layer of component nesting and eliminate CascadingAuthenticationState as public API.
-            builder.OpenComponent<CascadingAuthenticationState>(0);
-            builder.AddAttribute(1, nameof(CascadingAuthenticationState.ChildContent), (RenderFragment)(builder =>
+            if (ExistingCascadedAuthenticationState != null)
```

Comment:
This seems like it adds a fair bit of complexity to the world, do we have use cases in mind or is like this a better safe than sorry kind of thing?

Reply:
If the developer is cascading their own authentication state, they might be getting it from somewhere other than the `IAuthenticationStateProvider` in DI. It would be strange to overwrite their custom value with the output from `IAuthenticationStateProvider`.

I don't feel the extra complexity in the runtime code is significant at all. It does mean a couple of extra unit tests.

Reply:
Resolving because it didn't sound like you were massively concerned and I think the reasoning is good, but please let me know you think this is bad.